src/condor_tests/test_htcondor_annex_create_constraints.py

```diff
@@ -0,0 +1,224 @@
+    ( ['gpu-shared@expanse', '--cpus', '4', '--mem_mb', '1024', '--gpus', '5', '--project', 'project'], False ),
+    ( ['gpu-shared@expanse', '--cpus', '4', '--mem_mb', '1024', '--nodes', '1', '--gpus', '5', '--project', 'project'], False ),
+
+    ( ['gpu-shared@expanse', '--cpus', '80', '--mem_mb', '1024', '--gpus', '1', '--project', 'project'], False ),
+    ( ['gpu-shared@expanse', '--cpus', '4', '--mem_mb', str(1024*1024), '--gpus', '1', '--project', 'project'], False),
+    ( ['gpu-shared@expanse', '--cpus', '80', '--mem_mb', '1024', '--nodes', '1', '--gpus', '1', '--project', 'project'], False ),
+    ( ['gpu-shared@expanse', '--cpus', '4', '--mem_mb', str(1024*1024), '--nodes', '1', '--gpus', '1', '--project', 'project'], False),
+
+    # Verify that specifying a GPU type or a GPU count for a queue without
+    # GPUs fails.
+    ( ['normal@stampede2'], True ),
+    ( ['normal@stampede2', '--gpus', '1'], False ),
+    ( ['normal@stampede2', '--gpu-type', 'default'], False ),
+
+    # Verify that not specifying a GPU type, or specifying an invalid GPU
+    # type, for a GPU queue that requires one fails.
+    ( ['GPU@bridges2', '--gpus', 'v100-16:8'], True ),
+    ( ['GPU@bridges2', '--gpus', '8'], False ),
+    ( ['GPU@bridges2', '--gpus', 'not-a-type:8'], False ),
+    ( ['GPU@bridges2', '--gpus', '8', '--gpu-type', 'not-a-type'], False ),
+
+    # Verify that specifying a GPU type (with `:` or `--gpu-type`) for
+    # a GPU queue that doesn't have types fails.
+    ( ['gpu@anvil', '--nodes', '2', '--gpus', 'a_type:4', ], False ),
+    ( ['gpu@anvil', '--nodes', '2', '--gpus', '4', '--gpu-type', 'a_type' ], False ),
+
+    # Verify that queues respect max_duration.
+    ( ['normal@stampede2', '--nodes', '2', '--lifetime', str((48 * 60 * 60) - 1)], True ),
+    ( ['normal@stampede2', '--nodes', '2', '--lifetime', str((48 * 60 * 60) + 1)], False ),
+
+    ( ['development@stampede2', '--nodes', '2', '--lifetime', str((2 * 60 * 60) - 1)], True ),
+    ( ['development@stampede2', '--nodes', '2', '--lifetime', str((2 * 60 * 60) + 1)], False ),
+
+    ( ['wholenode@anvil', '--nodes', '2', '--lifetime', str((96 * 60 * 60) - 1)], True ),
+    ( ['wholenodel@anvil', '--nodes', '2', '--lifetime', str((96 * 60 * 60) + 1)], False ),
+
+    ( ['wide@anvil', '--nodes', '2', '--lifetime', str((12 * 60 * 60) - 1)], True ),
+    ( ['wide@anvil', '--nodes', '2', '--lifetime', str((12 * 60 * 60) + 1)], False ),
+
+    ( ['EM@bridges2', '--cpus', '24', '--lifetime', str((120 * 60 * 60) - 1)], True ),
+    ( ['EM@bridges2', '--cpus', '24', '--lifetime', str((120 * 60 * 60) + 1)], False ),
+]
+
+
+@pytest.fixture(params=TEST_CASES)
+def test_case(request):
+    return (request.param[0], request.param[1])
+
+
+@pytest.fixture
+def args(test_case):
+    return test_case[0]
+
+
+@pytest.fixture
+def expected(test_case):
+    return test_case[1]
+
+
+def test_hac(args, expected):
+    env = {** os.environ, '_CONDOR_HPC_ANNEX_ENABLED': 'TRUE',}
```

Comment:
```suggestion
    env = {**os.environ, '_CONDOR_HPC_ANNEX_ENABLED': 'TRUE',}
```
Python style: no space between the `**` and the variable name for dict expansion.